OData/src/System.Web.OData/Extensions/HttpRequestMessageProperties.cs

```diff
@@ -146,6 +165,12 @@ public Routing.ODataPath Path
                     return (long)totalCount;
                 }
 
+                if (this.TotalCountFunc != null)
+                {
+                    _request.Properties[TotalCountKey] = this.TotalCountFunc();
```

Comment:
Cache the result of `_request.Properties[TotalCountKey]` to a local variable to increase performance.